### ISSUE TITLE
Doesn't use the built-in TimeoutError

### ISSUE THREAD

**Issue by terrdavis** [pnpnpn/timeout-decorator#49]
Why don't you use the built-in exception?

**Comment by majiang** [comment]
The reason I think is that [built-in `TimeoutError` is "Raised when **a system function** timed out **at the system level**.](https://docs.python.org/3/library/exceptions.html#TimeoutError)

Besides, using the same name `TimeoutError` seems error-prone (that's why I searched the issues for the name and found this one).

**Comment by bfontaine** [comment]
It might be a good solution to use something like `class TimeoutDecoratorError(TimeoutError)`: a class with a different name that inherits from the built-in exception.

**Comment by majiang** [comment]
If you inherited you would say "a TimeoutDeoratorError **is a** TimeoutError" which is system level by built-in definition.  We should rather just rename the TimeoutError to a non-confusing alternative.

**Comment by bfontaine** [comment]
That’s true. `requests` uses [`Timeout`](https://github.com/kennethreitz/requests/blob/083aa67a0d3309ebe37eafbe7bfd96c235a019cf/requests/exceptions.py#L44) to avoid the confusion with the built-in `TimeoutError`.
